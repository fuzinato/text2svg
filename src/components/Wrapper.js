import React from 'react'
import { string} from 'prop-types';

const Wrapper = ({title, plusClass, children}) => {
  return (
    <div className={`input-section ${plusClass}`}>
      <span>{title}</span>
      {children}
    </div>
  )
}

Wrapper.propTypes = {
  title: string.isRequired,
  plusClass: string,
};

export default Wrapper
