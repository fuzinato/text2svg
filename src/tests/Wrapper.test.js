import React from 'react';
import Wrapper from '../components/Wrapper';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Wrapper ', () => {

  test('should have correct shapshot', () => {
    const component = renderer.create(
      <Wrapper title="Some Dummy Title" plusClass="some-dditional-class">Testing Wrapper</Wrapper>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should show correct title', () => {
    const componentDOM = shallow(
      <Wrapper title="Some Dummy Title" plusClass="some-dditional-class"></Wrapper>
    )
    expect(componentDOM.text()).toEqual('Some Dummy Title');
  });

  test('should correctly render children', () => {
    const componentDOM = shallow(
      <Wrapper title="" plusClass="some-dditional-class">Dummy Children</Wrapper>
    )
    expect(componentDOM.text()).toEqual('Dummy Children');
  });

  test('should have "input-section" in class', () => {
    const componentDOM = shallow(
      <Wrapper title="" />
    )
    expect(componentDOM.hasClass('input-section')).toEqual(true);
  });

  test('should have correct class name passed via plusClass prop', () => {
    const componentDOM = shallow(
      <Wrapper title="" plusClass="some-dditional-class" />
    )
    expect(componentDOM.hasClass('some-dditional-class')).toEqual(true);
  });
})