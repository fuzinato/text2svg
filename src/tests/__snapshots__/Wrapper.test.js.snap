// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper  should have correct shapshot 1`] = `
<div
  className="input-section some-dditional-class"
>
  <span>
    Some Dummy Title
  </span>
  Testing Wrapper
</div>
`;
